### DISCUSSION TITLE
Migrate from CRA to Next.js - After.js

### DISCUSSION THREAD

**Discussion by hiradary** [vercel/next.js#13443] [Ideas]
Hi!
I'm currently developing an enterprise e-commerce project, and I'm planning to implement SSR on it.
It first started off with CRA, now that It's grown so much, I'm thinking about ways I can get this SSR thing up & running. Next.js seems like a really cool tool! But I've also checked out After.js, which is kinda the same, but It allows you to use React Router. On the other hand, It definitely has fewer features and a much smaller community.
I'm wondering which way to go? Both have their own benefits

**Comment by stunaz** [comment]
Next.js for all the reason you mentioned.

I you want to keep up with RR, maybe wait for remix

**Comment by leerob** [comment]
Which routing features of React Router specifically do you find you can't achieve with Next.js? Or are you just already more familiar with that? 

I've seen people mention this, but I'm not as familiar with React Router as others, so just generally curious.

**Reply by benkingcode** [reply]
Next doesn't support nested routing, which means you can't really have nested layouts. Every page blows away the DOM. This is the biggest hard blocker for me adopting Next.

**Comment by revskill10** [comment]
You can use both CRA and NextJS in same project if you want.
CRA for internal admin routes and NextJS for public-facing routes.

**Reply by ApayRus** [reply]
I'm seeking for articles about it. But see only CRA vs. Next.js. 
Do you know where can I read about using both together?

**Comment by danilofuchs** [comment]
I recently made the transition from CRA to Next.js too, and it took much more work than we anticipated, but the benefits are immense! Specially if you value improved SEO performance.

Our site has localized routes, uses react-intl, styled-components, Typescript and is mostly static generated, except for redirect routes. This mix meant a lot of tweaking before achieving a successful transition.

Unfortunately, there is little content on this conversion at scale...

**Comment by gfortaine** [comment]
[Gradual Migration Strategies For Next.js](http://op.co.de/WfVUdF)
